docs/advanced-topics/message-system/rpc-params.md

```diff
@@ -63,11 +63,15 @@ struct ClientRpcParams
 }
 
 
-// Both ClientRPC methods below are fine, `ClientRpcParams` is completely optional
+// Both ClientRpc methods below are fine, `ClientRpcParams` is completely optional
 
 [ClientRpc]
 void AbcdClientRpc(int framekey) { /* ... */ }
 
 [ClientRpc]
-void XyzwClientRpc(int framekey, ClientRpcParams rpcParams = default) { /* ... */ }
-```
\ No newline at end of file
+void XyzwClientRpc(int framekey, ClientRpcParams clientRpcParams = default) { /* ... */ }
+```
+
+:::tip
+`ClientRpcSendParams`'s `TargetClientIds` property is a `ulong[]` which means everytime you try to specify a subset of target clients or even a single client target, you will have to allocate a `new ulong[]`. This pattern could quickly lead into lots of heap allocations and pressure GC which would cause GC spikes at runtime. We suggest developers to cache their `ulong[]` variables or use an array pool to cycle `ulong[]` instances so that it would cause less heap allocations.
+:::
```

Comment:
actually, this tip block was the main purpose of this PR but I ended up cleaning bunch of other stuff as well :)